client/layout/masterbar/masterbar-cart/masterbar-cart-button.tsx

```diff
@@ -0,0 +1,64 @@
+import { Popover } from '@automattic/components';
+import { CheckoutErrorBoundary } from '@automattic/composite-checkout';
+import { MiniCart } from '@automattic/mini-cart';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { useTranslate } from 'i18n-calypso';
+import { useRef, useState } from 'react';
+import MasterbarItem from '../item';
+
+import './masterbar-cart-button-style.scss';
+
+export type MasterbarCartButtonProps = {
+	selectedSiteSlug: string | undefined;
+	goToCheckout: ( siteSlug: string ) => void;
+};
+
+export function MasterbarCartButton( {
+	selectedSiteSlug,
+	goToCheckout,
+}: MasterbarCartButtonProps ): JSX.Element | null {
+	const { responseCart, reloadFromServer } = useShoppingCart( selectedSiteSlug );
+	const cartButtonRef = useRef( null );
+	const [ isActive, setIsActive ] = useState( false );
+	const translate = useTranslate();
+
+	if ( ! selectedSiteSlug || responseCart.products.length < 1 ) {
+		return null;
+	}
+
+	const onClick = () => {
+		setIsActive( ( active ) => {
+			if ( ! active ) {
+				reloadFromServer();
+			}
+			return ! active;
+		} );
+	};
+	const onClose = () => setIsActive( false );
+	const tooltip = translate( 'My shopping cart' );
+
+	return (
+		<div className="masterbar-cart-button" ref={ cartButtonRef }>
+			<CheckoutErrorBoundary errorMessage="Error">
+				<MasterbarItem icon="cart" tooltip={ tooltip } onClick={ onClick } />
+				<MasterbarCartCount productsInCart={ responseCart.products.length } />
```

Comment:
I've just noticed something - if I roll over on the count, it doesn't have the pointer cursor, nor anything happens if I click on the count. 

I expected that the count is part of the button, and if I click on it, it would work the same way as the button works.

Reply:
Good point! It turns out that's how the popover cart works too, but it's unnecessary here. I've reorganized the count so that it is inside the button.